java/java.editor/test/unit/src/org/netbeans/modules/editor/java/SyntaxColoringTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.editor.java;
+
+import javax.swing.JEditorPane;
+import javax.swing.SwingUtilities;
+import javax.swing.text.AbstractDocument;
+import javax.swing.text.Document;
+
+import org.netbeans.api.lexer.TokenHierarchy;
+import org.netbeans.api.lexer.TokenSequence;
+import org.netbeans.junit.NbTestCase;
+import org.netbeans.modules.editor.java.JavaKit;
+import org.openide.text.CloneableEditorSupport;
+
+public class SyntaxColoringTest extends NbTestCase {
+
+	private int tokenCount;
+
+	public SyntaxColoringTest(String name) {
+		super(name);
+	}
+
+	public void testTokenHierarchy() {
+		SwingUtilities.invokeLater(() -> {
+			JEditorPane pane = new JEditorPane();
+			pane.setEditorKit(CloneableEditorSupport.getEditorKit(JavaKit.JAVA_MIME_TYPE));
+			pane.setText("public class Hello {}");
+			AbstractDocument document = (AbstractDocument) pane.getDocument();
+			document.readLock();
+			TokenHierarchy<Document> tokenHierarchy = TokenHierarchy.get(document);
+			if (tokenHierarchy != null) {
+				TokenSequence<?> tokenSequence = tokenHierarchy.tokenSequence();
+				if (tokenSequence != null) {
+					tokenCount = tokenSequence.tokenCount();
+				}
+			}
+		});
+
+		try {
+			Thread.sleep(5000);
```

Comment:
lets try to implement this without `sleep`. 5s are long for a test but making it shorter will cause trouble under load.

e.g `invokeAndWait` or a semaphore would be an option.

Reply:
If the intention is, that the complete test is run on the EDT, `NbTestCase` can be told to do that:

http://bits.netbeans.org/dev/javadoc/org-netbeans-modules-nbjunit/org/netbeans/junit/NbTestCase.html#runInEQ--

return `true` from it and you are done.